webui/react/src/pages/F_ExpList/glide-table/tooltip.tsx

```diff
@@ -0,0 +1,93 @@
+import { DataEditorProps, GridMouseEventArgs } from '@glideapps/glide-data-grid';
+import { Tooltip } from 'antd';
+import React, { MutableRefObject, ReactNode, useCallback, useEffect, useMemo } from 'react';
+
+import { Loadable } from 'utils/loadable';
+
+import { ColumnDefs, ExperimentColumn } from './columns';
+import { GlideTableProps } from './GlideTable';
+
+export interface TooltipProps {
+  x: number;
+  y: number;
+  open: boolean;
+  text: string;
+}
+
+interface UseTooltipParams {
+  data: GlideTableProps['data'];
+  columnIds: ExperimentColumn[];
+  columnDefs: ColumnDefs;
+}
+interface UseTooltipReturn {
+  closeTooltip: () => void;
+  onItemHovered: DataEditorProps['onItemHovered'];
+  tooltip: ReactNode;
+}
+
+export const useTableTooltip = ({
+  data,
+  columnIds,
+  columnDefs,
+}: UseTooltipParams): UseTooltipReturn => {
+  const [tooltipProps, setTooltipProps] = React.useState<TooltipProps | undefined>(undefined);
+
+  const timeoutRef = React.useRef(0);
+  const closeTooltip = useCallback(() => {
+    window.clearTimeout(timeoutRef.current);
+    timeoutRef.current = 0;
+    setTooltipProps(undefined);
+  }, []);
+
+  const onItemHovered: DataEditorProps['onItemHovered'] = React.useCallback(
+    (args: GridMouseEventArgs) => {
+      if (args.kind === 'cell') {
+        window.clearTimeout(timeoutRef.current);
+        setTooltipProps(undefined);
+        const [columnIdx, rowIdx] = args.location;
+        const record = data[rowIdx];
+        if (record && Loadable.isLoaded(record)) {
+          const columnId = columnIds[columnIdx];
+          const text = columnDefs[columnId].tooltip(record.data);
+          if (text) {
+            timeoutRef.current = window.setTimeout(() => {
+              setTooltipProps({
+                open: true,
+                text: text,
+                x: args.bounds.x + 20,
+                y: args.bounds.y + 20,
+              });
+            }, 500);
```

Comment:
not necessarily something that needs to be done in this PR, but wondering if it might make sense to have some of the tooltip configuration (design, timeout period, which css variables to use) shared between the UI kit component and the grid version. also potentially for other elements that are also components in the UI kit (avatar? checkbox?). just so if we make changes to the UI kit components, we only have to do that in one place for the grid to also get updated. probably fine for now for the grid to be somewhat independent but just a potential improvement that may be worth it in some cases.